processing/src/main/java/org/apache/druid/segment/RowBasedCursor.java

```diff
@@ -140,4 +148,12 @@ private void advanceToMatchingRow()
       rowId++;
     }
   }
+
+  private void advanceToMatchingMarkRow()
+  {
+    while (!isDone() && rowId < markId) {
```

Comment:
This should be checking `valueMatcher` somehow, right? I think as written, it will advance `markId` number of rows on the underlying `rowWalker`, without regard for which ones match the provided filter. (Normally, `rowId` is incremented once for each row that matches the filter.)

Additionally, this logic creates a requirement that `rowWalker` be immutable. I don't think that'll always be true. For example, we use this cursor on top of external data, where re-walking the walker means re-fetching a file. The file could have changed. We also use it on top of lookups, where re-walking could get a different set of keys. I am not sure how really to do sensible marking on data streams like this.